types/luks.nix

```diff
@@ -65,6 +95,9 @@
         [
           # TODO do we need this always in initrd and only there?
           { boot.initrd.luks.devices.${config.name}.device = dev; }
+          (lib.optional (config.keyFile != null) { boot.initrd.luks.devices.${config.name} = {inherit (config) keyFile; };})
+          (lib.optional (config.keyFileSize != null) { boot.initrd.luks.devices.${config.name} = {inherit (config) keyFileSize; };})
+          (lib.optional (config.keyFileOffset != null) { boot.initrd.luks.devices.${config.name} = { inherit (config) keyFileOffset; };})
```

Comment:
hmm, i hadn't thought about this but i guess disko's `keyFile` option isn't meant to be the same as NixOS' right?

if you'd prefer to avoid making this assumption, i can just remove this commit & manually set these options in my NixOS config separately.